lm_eval/models/huggingface.py

```diff
@@ -190,6 +196,27 @@ def _create_auto_tokenizer(
         tokenizer.pad_token = tokenizer.eos_token
         return tokenizer
 
+    @property
+    def add_special_tokens(self) -> bool:
+        """Whether to include special tokens in encoded text. This should be
+        determined by whether or not the model was trained with special tokens.
+
+        TODO: Remove these conditionals once HuggingFace supports a way to
+        check whether or not an arbitrary model was trained with special tokens.
+        """
+        if self._add_special_tokens is not None:
+            return self._add_special_tokens
+        elif self.AUTO_MODEL_CLASS is transformers.AutoModelForCausalLM:
```

Comment:
Do all causal models not use special tokens (e.g., BOS)?

Reply:
No, but I believe the majority do not. More importantly, this is for backward compatibility as the [`_loglikelihood_tokens()`](https://github.com/bigscience-workshop/lm-evaluation-harness/blob/d02503639ed44d6f017a7b65848396d54a2998ee/lm_eval/api/model.py#L201) method for causal LMs relies on a no-special-tokens assumption given that contexts and labels/continuations are tokenized separately and then concatenated (see [here](https://github.com/bigscience-workshop/lm-evaluation-harness/blob/d02503639ed44d6f017a7b65848396d54a2998ee/lm_eval/api/model.py#L167)). This means we can't allow users to specify `add_special_tokens` for causal models at the moment. Thanks for pointing this out!

Reply:
I see, it might be important to note this in a comment